pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -1022,6 +1025,9 @@ public Response endReplaceSegments(
       Preconditions.checkNotNull(segmentLineageEntryId, "'segmentLineageEntryId' should not be null");
       _pinotHelixResourceManager.endReplaceSegments(tableNameWithType, segmentLineageEntryId,
           endReplaceSegmentsRequest);
+      if (pruneSegments) {
+        _pinotHelixResourceManager.invokeControllerPeriodicTask(tableNameWithType, RetentionManager.TASK_NAME, null);
```

Comment:
Invoking retention manager is too expensive and will touch all tables. We should pass this parameter into `_pinotHelixResourceManager.endReplicaSegments()` call and only delete the segments replaced for this table

Reply:
Not introduced in this PR, but we didn't check the return value of `_pinotHelixResourceManager.endReplicaSegments()`

Reply:
> Invoking retention manager is too expensive and will touch all tables.

Is that so? Passing the `tableNameWithType` parameter it would be expected to just affect the referred table.

> Not introduced in this PR, but we didn't check the return value of _pinotHelixResourceManager.endReplicaSegments()

If you meant `_pinotHelixResourceManager.endReplaceSegments()` it has no return value.